### PR TITLE
Feature #23408 feat: New configs to display featured contacts at the top & number of contacts to display on a contact list page.

### PR DESCRIPTION
**Pull request by vaibhavsTekdi**
### Is your feature request related to a problem? Please describe.

Yes, the user cannot able to see the featured contacts at the top in the contact list & also not able to set the number of contacts to display on a single page.

### Describe the solution you'd like

Add new configs are as follows :

- Contacts to List: Should have numbers to show contacts on list page with 'All' option.
- Sort By: Featured Contacts Ordering (It will display the featured contacts at the top in the contact list)

Add above configs about showing the number of contacts(default numbers) to display on contact list page & also add a new sort option as 'Featured Contacts Ordering' in sort by config under 'Contacts: Options >> List Layouts' tab as global at the component level.

Also, these configs should available at Joomla contact menu having 'List Contacts in a Category' menu type. Both the configs should have global values as default values. 

Contacts list should display according to the contact menu level configs.

### Additional context

With the existing config(s) at Contacts component level options & Contacts menu level, the user cannot able to see the featured contacts at the top in the contact list & also not able to set the number of contacts to display on a single page. 
It is showing the contacts sort by name or ordering or sort name etc.

### CODE REVIEW

File: administrator/language/en-GB/en-GB.ini [hunk 1]
```diff
@@ -494,6 +494,8 @@ JGLOBAL_NUMBER_CATEGORY_ITEMS_DESC="If Show, the number of articles in the categ
 JGLOBAL_NUMBER_CATEGORY_ITEMS_LABEL="Show Article Count"
 JGLOBAL_NUMBER_ITEMS_LIST_DESC="Default number of articles to list on a page."
 JGLOBAL_NUMBER_ITEMS_LIST_LABEL="# Articles to List"
+JGLOBAL_NUMBER_CONTACTS_LIST_DESC="Default number of contacts to list on a page."
```
Comment (Quy) [inline administrator/language/en-GB/en-GB.ini]: Please sort keys in alpha order.

**Comment by infograf768** [comment]
Please modify constants below to `COM_CONTACT_etc` and move both strings to the com_contact.ini file
Articles similar strings should never have been created as JGLOBAL_and placed in en-GB.ini. 

JGLOBAL_NUMBER_CONTACTS_LIST_DESC="Default number of contacts to list on a page."
JGLOBAL_NUMBER_CONTACTS_LIST_LABEL="# Contacts to List"

File: administrator/language/en-GB/en-GB.ini (hunk 1 above)
Comment (vaibhavsTekdi) [reply administrator/language/en-GB/en-GB.ini]: Moved these constants to the com_contact.ini file

**Comment by vaibhavsTekdi** [comment]
> Please modify constants below to `COM_CONTACT_etc` and move both strings to the com_contact.ini file
> Articles similar strings should never have been created as JGLOBAL_and placed in en-GB.ini.
> 
> JGLOBAL_NUMBER_CONTACTS_LIST_DESC="Default number of contacts to list on a page."
> JGLOBAL_NUMBER_CONTACTS_LIST_LABEL="# Contacts to List"

Modified both the constants & moved to the com_contact.ini file.

**Comment by infograf768** [comment]
`_N_ITEMS` in the constants makes no sense here as they are Label and Description which do not use a sprintf and variable.
Please just use
label="COM_CONTACT_NUMBER_CONTACTS_LIST_LABEL"
description="COM_CONTACT_NUMBER_CONTACTS_LIST_DESC"

**Comment by vaibhavsTekdi** [comment]
> `_N_ITEMS` in the constants makes no sense here as they are Label and Description which do not use a sprintf and variable.
> Please just use
> label="COM_CONTACT_NUMBER_CONTACTS_LIST_LABEL"
> description="COM_CONTACT_NUMBER_CONTACTS_LIST_DESC"

Fixed.

**Comment by uthorat** [comment]
I have tested this item :white_check_mark: successfully on 6ab994aaeb468dd75f557ba94f388f5b9b8a156e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23409">issues.joomla.org/tracker/joomla-cms/23409</a>.</sub>

**Comment by tushs24** [comment]
I have tested this item :white_check_mark: successfully on 6ab994aaeb468dd75f557ba94f388f5b9b8a156e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23409">issues.joomla.org/tracker/joomla-cms/23409</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23409">issues.joomla.org/tracker/joomla-cms/23409</a>.</sub>

**Comment by pallaviu** [comment]
I have tested this item :white_check_mark: successfully on 6ab994aaeb468dd75f557ba94f388f5b9b8a156e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23409">issues.joomla.org/tracker/joomla-cms/23409</a>.</sub>

**Comment by rdeutz** [comment]
@wilsonge rebase to 4.0-dev?

**Comment by shraddhaSankpal27** [comment]
I have tested this item :white_check_mark: successfully on 6ab994aaeb468dd75f557ba94f388f5b9b8a156e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23409">issues.joomla.org/tracker/joomla-cms/23409</a>.</sub>

**Comment by punambaravkar** [comment]
I have tested this item :white_check_mark: successfully on 2009b15af1bedd5c26ede7c1a117380569fbb68e<br /><br />## config related to pagination works fine <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23409">issues.joomla.org/tracker/joomla-cms/23409</a>.</sub>

**Comment by KishoriBKarale** [comment]
I have tested this item :white_check_mark: successfully on 2009b15af1bedd5c26ede7c1a117380569fbb68e<br /><br />Updated code about pagination & this feature working fine.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23409">issues.joomla.org/tracker/joomla-cms/23409</a>.</sub>

**Comment by uthorat** [comment]
I have tested this item :white_check_mark: successfully on 2009b15af1bedd5c26ede7c1a117380569fbb68e<br /><br />Checked with updated code about pagination and its working fine.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23409">issues.joomla.org/tracker/joomla-cms/23409</a>.</sub>

**Comment by HLeithner** [comment]
@vaibhavsTekdi please rebase this feature to 4.0 and create a new PR, this PR will likely not go into 3.9

**Comment by vaibhavsTekdi** [comment]
Rebased this feature to 4.0. **Check the new PR link [here.](https://github.com/joomla/joomla-cms/pull/23710)**

**Pull request closed without merging**